### ISSUE TITLE
Deno Debug Terminal

### ISSUE THREAD

**Issue by felipecrs** [denoland/vscode_deno#885]
**Is your feature request related to a problem? Please describe.**

VS Code has the embedded JavaScript Debug Terminal for debugging Node.js applications. It's extremely handy, as you don't need to worry about writing `launch.json` again.

Especially for CLI applications where you need to test your application under different scenarios like different arguments or even environment variables.

To aid with the arguments testing, you can do something like I did on https://github.com/teaxyz/cli/pull/669, but it's a lot more painful than just issuing commands on the terminal and having VS Code automatically pick it up on debug.

**Describe the solution you'd like**

A Deno Debug Terminal, pretty much like JavaScript Debug Terminal:

![image](https://github.com/denoland/vscode_deno/assets/29582865/ca4a33a9-f8c5-43c5-a9ac-39b3a4f68bdb)

https://github.com/denoland/vscode_deno/assets/29582865/9c403092-f678-4b24-8517-1a3939124804

**Comment by felipecrs** [comment]
I don't know if you know how useful this feature is. I am a Deno advocate, and I want to use it everywhere.

But it's hard (not to say impossible) to convince my team we would lose this feature, which would make debugging hundred times more complicated and slower.

**Comment by felipecrs** [comment]
Bun VS Code extension has added this feature: https://github.com/oven-sh/bun/pull/15801

It doesn't seem that complicated. I would be very grateful if some maintainer could at least give it a try.

**Comment by stefnotch** [comment]
Yes and no. I don't know what the security implications of always reading a `DENO_INSPECT` environment variable are.

It would be great to get an authoritative answer from someone on the Deno team.

**Comment by felipecrs** [comment]
Got it. Worst case, Deno could invoke the approval prompt. But I think it's overthinking indeed. 😅

**Comment by stefnotch** [comment]
So I guess the correct move now is to make the simplest possible PR, and ask about it there.

**Comment by stefnotch** [comment]
Ah, `BUN_INSPECT` isn't enough. That just tells Bun which port or whatever to use.

The second part is telling VSCode to attempt to connect to Bun. For that, the VSCode extension seems to start a [Tcp server/Unix socket](https://github.com/oven-sh/bun/blob/de487713ed48e8b170c7014534864aa61de212a2/packages/bun-debug-adapter-protocol/src/debugger/adapter.ts#L2187) and waits for someone to send a packet to it. The address of that server is in `BUN_INSPECT_NOTIFY`

**Comment by stefnotch** [comment]
Oh, the upstream issue is https://github.com/denoland/deno/issues/19328
